### PR TITLE
oil: update to 0.23.0

### PR DESCRIPTION
**Pull request by bugcrazy**
#### Testing the changes
- I tested the changes in this PR: **YES**

### CODE REVIEW

File: srcpkgs/oils/template [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
 makedepends="$hostmakedepends"
-short_desc="Oil, a new unix shell"
+short_desc="Oils, a new unix shell"
 maintainer="MarcoAPC <marcoaureliopc@gmail.com>"
 license="Apache-2.0"
 homepage="https://www.oilshell.org"
 distfiles="https://www.oilshell.org/download/oil-${version}.tar.xz"
 checksum=452e7fd428b686aa752cfeb9be2b94f9fcf7ba9ab22e0ae570a87cdcf782242e
-register_shell="/usr/bin/osh /usr/bin/oil"
+replaces="oil<=0.23.0"
```
Comment (classabbyamp) [inline srcpkgs/oils/template:15]: should be a transitional subpackage not replaces

Comment (ahesford) [reply srcpkgs/oils/template:15]: I think both replace and a transitional subpackage are appropriate here.

Comment (bugcrazy) [reply srcpkgs/oils/template:15]: I consider it more appropriate to replace, than transitional subpackage, only the names were changed, it is still the same model.

Comment (classabbyamp) [reply srcpkgs/oils/template:15]: without the transitional package, users would be stuck on the old package forever if they don't know to install the other one

Comment (bugcrazy) [reply srcpkgs/oils/template:15]: Done!

**Comment by leahneukirchen** [comment]
So, while we are renaming the package, should we do it like upstream suggests? https://github.com/oils-for-unix/oils/wiki/Oils-Packaging-Guidelines

**Comment by bugcrazy** [comment]
> So, while we are renaming the package, should we do it like upstream suggests? https://github.com/oils-for-unix/oils/wiki/Oils-Packaging-Guidelines

The definition itself is confusing, as there are two packages, one Python and one C, we should adopt more logical standards, this oils standard for package, is more correct.

**Comment by leahneukirchen** [comment]
Okay, so this is the old Python-version, but there upstream says:

> The oil tarball is the slow reference interpreter based on CPython.
>
>    Don't rename old distro packages. They can remain oil.
>    It's not worth packaging oil in new distros. It may be removed at some point.

So let's keep this (just adding the link is fine), and perhaps someone can contribute `oils-for-unix` instead?

**Comment by bugcrazy** [comment]
> So, while we are renaming the package, should we do it like upstream suggests? https://github.com/oils-for-unix/oils/wiki/Oils-Packaging-Guidelines

The definition itself is confusing, as there are two packages, one Python and one C, we should adopt more logical standards, this oils standard for package, is more correct.

> Okay, so this is the old Python-version, but there upstream says: https://www.oilshell.org/release/0.23.0/
> 
> > The oil tarball is the slow reference interpreter based on CPython.
> > Don't rename old distro packages. They can remain oil.
> > It's not worth packaging oil in new distros. It may be removed at some point.
> 
> So let's keep this (just adding the link is fine), and perhaps someone can contribute `oils-for-unix` instead?

Python version is slow, a few seconds, C++ version is slower than bash version.

Python version is more stable and tested, has a history of having fewer bugs.

Here is a speed test and some bugs of the C++ version:

https://github.com/oils-for-unix/oils/issues/1069

**Comment by ahesford** [comment]
We don't need multiple versions of this package in Void, and the project author notes that the "reference" implementation might go away at some point, so just pick one to be the version packaged by Void.

**Comment by leahneukirchen** [comment]
@ahesford doesnt matter, the new version should have a new package name anyway (and conflict, I guess)

**Comment by meator** [comment]
> and perhaps someone can contribute `oils-for-unix` instead?

I've tried to package `oils-for-unix`: #52218

**Comment by meator** [comment]
Also, I vote for keeping `oil` as is and not renaming it to `oils`. This is what upstream recommends.

**Comment by leahneukirchen** [comment]
yes, please keep the old name.

**Comment by bugcrazy** [comment]
> yes, please keep the old name.

Done!

**Comment by classabbyamp** [comment]
what will happen for anyone using `oil` as login shell?

**Comment by bugcrazy** [comment]
> what will happen for anyone using `oil` as login shell?
It will work normally.

**Pull request merged**